src/property-filter/token-editor.tsx

```diff
@@ -232,7 +215,7 @@ export function TokenEditor({
         ? temporaryToken.operator
         : allowedOperators[0];
     const matchedProperty = filteringProperties.find(property => property.propertyKey === newPropertyKey) ?? null;
-    setTemporaryToken({ ...temporaryToken, property: matchedProperty, operator });
+    setTemporaryToken({ ...temporaryToken, property: matchedProperty, operator, value: null });
```

Comment:
do we need a test for the new behaviour?

Reply:
It is tested on this line: https://github.com/cloudscape-design/components/pull/2437/files#diff-6b1307912e8d64fe1141424ab092e4961250328ee690ba1821fca17bcb32aa86R713